app/jobs/background_inactive_email_job.rb

```diff
@@ -1,6 +1,20 @@
 class BackgroundInactiveEmailJob < ApplicationJob
   def perform(user)
     return false if user.repo_subscriptions.present?
-    UserMailer.poke_inactive(user: user).deliver_later
+
+    with_most_issues = curated_repos_for(user).order_by_issue_count.first
```

Comment:
To me, overall this logic feels kinda complicated for a job. I'm not super familiar with this codebase, but something like a method on the `Repo` model or a `CuratedRepoService` would be how I would approach this - I like to keep my jobs pretty slim though.

Thoughts?

Reply:
The logic itself is already encapsulated inside the scopes and we're just chaining them.

`#curated_repos_for` is just the result of extracting duplicating code into a method to improve readability.

Not only replacing these calls by a service object call will add another layer of indirection but also it'll not be used by any other part of the code. I think we should wait to need the same behavior in a different place before extracting it.